Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorApplyMotionSetComponent.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#if defined(CARBONATED)
+#include <AzCore/std/smart_ptr/unique_ptr.h>
+#include <AzCore/Script/ScriptProperty.h>
+
+#include <AzToolsFramework/ToolsComponents/EditorComponentBase.h>
+
+#include <Integration/Components/ApplyMotionSetComponent.h>
+
+namespace EMotionFX
+{
+    namespace Integration
+    {
+        class EditorApplyMotionSetComponent
+            : public AzToolsFramework::Components::EditorComponentBase
+            , private AZ::Data::AssetBus::MultiHandler
+            , private EditorApplyMotionSetComponentRequestBus::Handler
+        {
+        public:
+
+            AZ_EDITOR_COMPONENT(EditorApplyMotionSetComponent, "{2734A694-3E28-46B4-9917-342FDA60BC0E}");
+
+            EditorApplyMotionSetComponent();
+            ~EditorApplyMotionSetComponent() override;
+
+            //////////////////////////////////////////////////////////////////////////
+            // AZ::Component interface implementation
+            void Activate() override;
+            void Deactivate() override;
+
+            //////////////////////////////////////////////////////////////////////////
+            // AZ::Data::AssetBus::Handler
+            void OnAssetReady(AZ::Data::Asset<AZ::Data::AssetData> asset) override;
+            void OnAssetReloaded(AZ::Data::Asset<AZ::Data::AssetData> asset) override;
+            //////////////////////////////////////////////////////////////////////////
+
+            //////////////////////////////////////////////////////////////////////////
+            // EditorAnimGraphComponentRequestBus::Handler
+            const AZ::Data::AssetId& GetMotionSetAssetId() override;
+
+            //////////////////////////////////////////////////////////////////////////
+            static void GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+            {
+                ApplyMotionSetComponent::GetProvidedServices(provided);
+            }
+
+            static void GetIncompatibleServices(AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+            {
+                ApplyMotionSetComponent::GetIncompatibleServices(incompatible);
+            }
+
+            static void GetDependentServices(AZ::ComponentDescriptor::DependencyArrayType& dependent)
+            {
+                ApplyMotionSetComponent::GetDependentServices(dependent);
+            }
+
+            static void GetRequiredServices(AZ::ComponentDescriptor::DependencyArrayType& required)
+            {
+                ApplyMotionSetComponent::GetRequiredServices(required);
+            }
+
+            static void Reflect(AZ::ReflectContext* context);
+            //////////////////////////////////////////////////////////////////////////
+
+        private:
+#if defined(CARBONATED)

```

Comment:
Looks like inner #ifdef/#endif for CARBONATED are not necessary
here and below
due to we use #if defined(CARBONATED) for whole file in the line 3
?